pkg/mimir/client/http_client.go

```diff
@@ -126,5 +128,42 @@ func createHTTPClient() (*http.Client, error) {
 		}
 		timeout = time.Duration(timeoutSeconds) * time.Second
 	}
-	return &http.Client{Timeout: timeout}, nil
+
+	tlsConfig := &tls.Config{}
+	httpClient := http.Client{
+		Timeout: timeout, 
+		Transport: &http.Transport{TLSClientConfig: tlsConfig},
+	}
+
+	if c.config.TLS.CAPath != "" {
+		certPool, err := x509.SystemCertPool()
+		if err != nil {
+			return nil, err
+		}
+
+		caCertPEM, err := os.ReadFile(c.config.TLS.CAPath)
+		if err != nil {
+			return nil, err
+		} 
+		
+		ok := certPool.AppendCertsFromPEM(caCertPEM)
+		if !ok {
+			return nil, err
+		}
+		
+		tlsConfig.RootCAs = certPool
+	}
+
+	if c.config.TLS.ClientCertPath != "" || c.config.TLS.ClientKeyPath != "" {
```

Comment:
Does it work if we only set one of them? In that case, should it be `&&`?

Reply:
`LoadX509KeyPair` should return an error if the file path doesn't exist and if the contents are not a valid certificate, is this enough or should I return an error when only one of them is configured?

Reply:
I think that it's enough. Thanks for the clarification.